File: scripts/check_csv.py
# Script that checks the csv file for the correct format
# Usage: python check_csv.py <csv_file>
# Example: python check_csv.py data.csv

# import libraries
import numpy as np
import pandas as pd
import sys

# set errors to false
errors = False

# read in the csv file
csv_file = sys.argv[1]

# read in the csv file
df = pd.read_csv(csv_file)

# print column names
print("Column names:")
print(df.columns)

# convert column names to lower case
df.columns = [x.lower() for x in df.columns]

# retain on





# check whether genome	type	start	end	strand	p_value	gene_name are present in the csv file.
if 'genome' in df.columns and 'type' in df.columns and 'start' in df.columns and 'end' in df.columns and\
        'strand' in df.columns and 'p_value' in df.columns and 'gene_name' in df.columns:
    print("All required columns are present")
    # change p_value column name to p_value_[filename]
    # get filename without .csv and path
    csv_filename = csv_file.split('/')[-1].replace('.csv', '')
    print(csv_filename)
    df = df.rename(columns={'p_value': 'p_value_' + csv_filename})
    # retain only "CDS" and "sRNA" in the "type" column
    df = df[df['type'].isin(["CDS", "sRNA"])]
    # print all types
    print("All types:")
    print(df['type'].unique())
    # save the csv file
    df.to_csv(csv_file, index=False)
else:
    # write error message with name of the csv file
    error_message = "Error: Missing required columns in the csv file: " + csv_file + "\n" + \
                    "Required columns are: genome, type, start, end, strand, p_value, gene_name"
    errors = True

    # raise an error
    raise ValueError(error_message)

File: scripts/check_csv.R
# Script that reads in file from Rscript call, and renames all the columns
# as the filename_ + column name to avoid conflicts with other files.
# it also checks whether the columns "p_value" and "gene_name" are present. If not, it will
# cause an error as output of the script.

# load packages
library(tidyverse)


# import the csv file name from the bash Rscript call
args <- commandArgs(trailingOnly = TRUE)
csv_file <- args[1]
# csv_file <- "./data/MAPS_test.csv"

# extract the filename from the csv file without .csv
name_output <- gsub(".csv", "", basename(csv_file))

# read in the csv file
csv_file <- read_csv(csv_file)

# check if the columns p_value and gene_name are present
if (!("p_value" %in% colnames(csv_file)) | !("gene_name" %in% colnames(csv_file))) {
  stop("Columns 'p_value' and 'gene_name' are not present in the file.")
} else {
  print("Columns 'p_value' and 'gene_name' are present in the file.")
}


# rename all the columns to avoid conflicts with other files, except for gene_name
for (col in colnames(csv_file)) {
  if (col != "gene_name") {
    colnames(csv_file)[colnames(csv_file) == col] <- paste0(name_output, "_", col)
  }
}


# rename all "-" and " " to "_" in the column names
colnames(csv_file) <- gsub("-", "_", colnames(csv_file))
colnames(csv_file) <- gsub(" ", "_", colnames(csv_file))

# write the tibble to a file
write_tsv(csv_file, paste0(name_output, ".tsv"))

# print the head of the tibble
head(csv_file)
